src/services/character/character.service.ts

```diff
@@ -1,31 +1,50 @@
-import _ from 'lodash'
-import { Document } from 'mongoose'
-import { ICharacter } from '../../types/character/character'
-import characters from '../../data/models/character-schema'
-
-export class CharacterService {
-  private static _instance: CharacterService
-
-  public static getInstance(): CharacterService {
-    if(_.isNil(CharacterService._instance)) {
-      CharacterService._instance = new CharacterService()
-    }
-    return CharacterService._instance
-  }
-
-  public async getEntity(ownerId: string):Promise<Document | void> {
-    return await characters.findOne({ ownerId }, (err: Error, characterFound: Document): ICharacter => {        
-      if (characterFound) return characterFound.schema.obj
-    })
-    .catch((err): void => {
-      console.log(`Error: Character not found because: ${err}`)
-      return
-    })
-  }
-
-  public setEntity(ownerId: string, characterToInsert: ICharacter): void {
-    characters.update({ ownerId }, characterToInsert, (err: Error): void => {
-      if (err) throw err
-    })
-  }
-}
+import _ from "lodash";
+import mongoose from "mongoose";
+import { ICharacter } from "../../types/character/character";
+
+export class CharacterService {
+  private static _instance: CharacterService;
+
+  public static getInstance(): CharacterService {
+    if (_.isNil(CharacterService._instance)) {
+      CharacterService._instance = new CharacterService();
+    }
+    return CharacterService._instance;
+  }
+
+  public getEntity(ownerId: string): Promise<ICharacter | void> {
+    return mongoose
+      .model(`characterSchema`)
+      .findOne({ ownerId })
+      .then((foundCharacter: mongoose.Document | null): void => {
+        if (this._isEntity(foundCharacter)) {
+          console.log(`Character found with name: ${foundCharacter.name}`);
+
+          void Promise.resolve(foundCharacter);
+        } else {
+          void Promise.reject(

```

Comment:
Nope the void is a really bad place here.
You MUST return the Promise to make your code works.

Reply:
Same as above